### PR TITLE
add deploy python scripts for text cls example

### LINKED ISSUE #278: python API模型推理能否增加NLP案例
在export model 之后，通过infference加载静态图模型，可以增加一个学习的案例吗？

### PR DESCRIPTION
**Pull request by Steffy-zxf**
### PR types

New features

### PR changes

Examples

### Describe

add deploy python scripts for text cls example (rnn & ptms)

### CODE REVIEW

File: examples/text_classification/rnn/deploy/python/predict.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    predictor = Predictor(args.model_file, args.params_file, args.device,
+                          args.max_seq_length)
+
+    # Firstly pre-processing prediction data  and then do predict.
+    data = [
+        '这个宾馆比较陈旧了，特价的房间也很一般。总体来说一般',
+        '怀着十分激动的心情放映，可是看着看着发现，在放映完毕后，出现一集米老鼠的动画片',
+        '作为老的四星酒店，房间依然很整洁，相当不错。机场接机服务很好，可以在车上办理入住手续，节省时间。',
+    ]
+    vocab = ppnlp.data.Vocab.load_vocabulary(
```
Comment (ZeyuChen) [inline examples/text_classification/rnn/deploy/python/predict.py]: 改为`from paddlenlp.data import Vocab`
`Vocab.load_vocabulary`

**Review by ZeyuChen — COMMENTED** [review]
补充测试下，对比下与原生python和paddleinference的速度，确保使用预测库的速度有提升。
同步对比下CPU和GPU。

File: examples/text_classification/pretrained_models/README.md [hunk 2]
```diff
@@ -138,10 +141,16 @@ checkpoints/
...
-$ python export_model.py --params_path=./checkpoint/model_900/model_state.pdparams --output_path=./static_graph_params
+python export_model.py --params_path=./checkpoint/model_900/model_state.pdparams --output_path=./static_graph_params
 ```
 其中`params_path`是指动态图训练保存的参数路径，`output_path`是指静态图参数导出路径。
 
+导出模型之后，可以用于部署，deploy/python/predict.py文件提供了python部署预测示例。运行方式：
+
+```shell
+python deploy/python/predict.py --model_file=static_graph_params.pdmodel --params_file=static_graph_params.pdiparams
+```
```
Comment (wawltor) [inline examples/text_classification/pretrained_models/README.md:152]: 这里的path，是否要和上面导出的路径一起使用，还是导出之后直接会是static_graph_params.pdmodel

File: examples/text_classification/pretrained_models/deploy/python/predict.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+
+from scipy.special import softmax
+from paddle import inference
+import numpy as np
+import paddle
```
Comment (wawltor) [inline examples/text_classification/pretrained_models/deploy/python/predict.py:18]: 这里的import关系有点乱
import numpy as np
import paddle
from paddle import inference

File: examples/text_classification/pretrained_models/deploy/python/predict.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+            # set GPU configs accordingly
+            config.enable_use_gpu(100, 0)
+        elif device == "cpu":
+            # set CPU configs accordingly,
+            # such as enable_mkldnn, set_cpu_math_library_num_threads
+            config.disable_gpu()
+        elif device == "xpu":
+            # set XPU configs accordingly
+            config.enable_xpu(100)
+        config.switch_use_feed_fetch_ops(False)
```
Comment (wawltor) [inline examples/text_classification/pretrained_models/deploy/python/predict.py:107]: 这里的超参 100是什么意思了？

File: examples/text_classification/rnn/deploy/python/predict.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        args.vocab_path, unk_token='[UNK]', pad_token='[PAD]')
+    tokenizer = JiebaTokenizer(vocab)
+
+    data = [
+        '这个宾馆比较陈旧了，特价的房间也很一般。总体来说一般',
+        '怀着十分激动的心情放映，可是看着看着发现，在放映完毕后，出现一集米老鼠的动画片',
+        '作为老的四星酒店，房间依然很整洁，相当不错。机场接机服务很好，可以在车上办理入住手续，节省时间。',
+    ]
```
Comment (wawltor) [inline examples/text_classification/rnn/deploy/python/predict.py]: 这块的data是否重复了

File: examples/text_classification/rnn/deploy/python/predict.py [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+
+from scipy.special import softmax
+from paddle import inference
+import numpy as np
+import paddle
+
```
Comment (wawltor) [inline examples/text_classification/rnn/deploy/python/predict.py]: 同上

File: examples/text_classification/rnn/deploy/python/predict.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+        results = []
+        for batch in batches:
+            input_ids, seq_lens = batchify_fn(batch)
+            self.input_handles[0].copy_from_cpu(input_ids)
+            if network in [
+                    "lstm", "bilstm", "gru", "bigru", "rnn", "birnn",
+                    "bilstm_attn"
+            ]:
+                self.input_handles[1].copy_from_cpu(seq_lens)
+            self.predictor.run()
```
Comment (wawltor) [inline examples/text_classification/rnn/deploy/python/predict.py:134]: 这里如果是network是RNN类就copy是什么意思了？

File: examples/text_classification/rnn/deploy/python/predict.py (hunk 1 above)
Comment (Steffy-zxf) [reply examples/text_classification/rnn/deploy/python/predict.py]: Done

File: examples/text_classification/pretrained_models/README.md (hunk 2 above)
Comment (Steffy-zxf) [reply examples/text_classification/pretrained_models/README.md:152]: 在文档中指定的导出和加载路径是一致的

File: examples/text_classification/pretrained_models/deploy/python/predict.py (hunk 3 above)
Comment (Steffy-zxf) [reply examples/text_classification/pretrained_models/deploy/python/predict.py:18]: Done

File: examples/text_classification/pretrained_models/deploy/python/predict.py (hunk 4 above)
Comment (Steffy-zxf) [reply examples/text_classification/pretrained_models/deploy/python/predict.py:107]: 表示预先分配的显存，paddle inference api示例给的是100

File: examples/text_classification/rnn/deploy/python/predict.py (hunk 5 above)
Comment (Steffy-zxf) [reply examples/text_classification/rnn/deploy/python/predict.py]: Done

File: examples/text_classification/rnn/deploy/python/predict.py (hunk 6 above)
Comment (Steffy-zxf) [reply examples/text_classification/rnn/deploy/python/predict.py]: Don

File: examples/text_classification/rnn/deploy/python/predict.py (hunk 7 above)
Comment (Steffy-zxf) [reply examples/text_classification/rnn/deploy/python/predict.py:134]: 因为RNN类网络还有一个输入是seq len。如果是rnn类网络则需要增加一个seq len输入，如果是bow网络，则不需要。

**Review by wawltor — APPROVED** [review]
LGTM
